iothub/device/src/Transport/DeviceIdentity.cs

```diff
@@ -20,11 +20,15 @@ internal class DeviceIdentity
         internal ProductInfo ProductInfo { get; }
         internal AuthenticationModel AuthenticationModel { get; }
         internal string Audience { get; }
-        internal DeviceIdentity(IotHubConnectionString iotHubConnectionString, AmqpTransportSettings amqpTransportSettings, ProductInfo productInfo)
+        internal ClientOptions Options { get; }
+
+        internal DeviceIdentity(IotHubConnectionString iotHubConnectionString, AmqpTransportSettings amqpTransportSettings, ProductInfo productInfo, ClientOptions options)
         {
```

Comment:
I wish product info can be a part of client options evry where. I know that is a breaking change; but is it possible to make it so in implementation except for public API?

Reply:
Yeah, we could. 

What do you think about marking `DeviceClient.ProductInfo` as deprecated, and adding `ProductInfo` into `ClientOptions`?

Reply:
I would love that but I think we should hold off on breaking changes at this point. So just internal implementation should be good for now

Reply:
This requires a bit of refactoring, I'll add this in a subsequent PR.